src/app/modules/widgets/tree-viewer/tree-viewer.component.ts

```diff
@@ -19,6 +20,21 @@ export class TreeViewerComponent implements OnInit {
   @Input() level = 0;
   @Input() config: TreeViewerConfig;
 
+  _newElementConfig: NewElementConfig;
+
+  get newElementConfig(): NewElementConfig {
+    if (!this._newElementConfig) {
+      this._newElementConfig = {
+        indent: this.isLeaf ? '0px' : '12px',
+        context: { node: this.model, type: this.isLeaf ? ContextType.Sibling : ContextType.Parent},
+        iconCssClasses: this.model.leafCssClasses,
```

Comment:
how about `this.config.cssClasses ? this.config.cssClasses : this.model.leafCssClasses`?
The icon should differ for file/folder creation (depending on whatever node types the parent defines).

Reply:
The css classes are provided via msg bus event payload now, too.